app/main.py

```diff
@@ -1,13 +1,39 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+    return {
+        "line": error["line_number"], 
+        "column": error["column_number"], 
+        "message": error["text"], 
+        "name": error["code"], 
+        "source": "flake8"
+        }
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+    return {
+        "errors": 
+            [{"line": error["line_number"], 
+                "column": error["column_number"], 
+                "message": error["text"], 
+                "name": error["code"], 
+                "source": "flake8"} for error in errors
+            ],  
+        "path": file_path, 
+        "status": "failed"
+        }
 
 
 def format_linter_report(linter_report: dict) -> list:
-    # write your code here
-    pass
+    return[
+        {"errors": [], "path": list(linter_report.keys())[i], "status": "passed"} if i == 0 else
```

Comment:
The logic here sets the status to 'passed' only for the first file in the linter report. If the intention is to mark files as 'passed' only when there are no errors, you might need to adjust this logic to check for the presence of errors instead of relying on the index `i`.